ckanext/opendatani/controller.py

```diff
@@ -249,3 +254,57 @@ def resource_read(self, id, resource_id):
                     h.flash_error('Sorry, this file is too large to be able to display in the browser, please download the data resource to examine it further.')
         template = self._resource_template(dataset_type)
         return render(template, extra_vars=vars)
+
+
+class CustomReportController(CorePackageController):
+    def prepare_report(self, org):
+        """
+        Creates a CSV publisher report and returns it as a string
+        :param org: organization
+        :type org: string
+        :return: a CSV string
+        :rtype: string
+        """
+
+        try:
+            data_dict = {'org_name': org}
+
+            if org == 'full-portal':
```

Comment:
`org` is pulled from the URL name of the organization. It was the only way I found to pass a variable to the Jinja template to be able to call a controller on demand. I have found now that you can't create a URL with any special characters other than '-'. So I decided now to pass back from the full report template something like `@full` for the sake of clarity (though it could actually be a single special character, since a URL with one can never exist).